### PR TITLE
Define api

### PR DESCRIPTION
**Pull request by wistefan**

### CODE REVIEW

File: api/trusted-issuers-list.yaml [hunk 1]
```diff
@@ -108,5 +139,27 @@ components:
             - 12
             - "GOLD_CUSTOMER"
             - "STANDARD_CUSTOMER"
```
Comment (pulledtim) [inline api/trusted-issuers-list.yaml:141]: Why did you remove the reference for an embedded object here? The code still allows it or?

File: src/main/java/org/fiware/iam/TILMapper.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+	default Claim map(ClaimVO claimVO) {
+		return new Claim()
+				.setName(claimVO.getName())
+				.setClaimValues(
+						claimVO.getAllowedValues().stream().map(value -> {
+									try {
+										return OBJECT_WRITER.writeValueAsString(value);
+									} catch (JsonProcessingException e) {
+										return null;
```
[the remark this answers is not in the document]
Comment (wistefan) [reply src/main/java/org/fiware/iam/TILMapper.java:57]: Yes, my inability to properly inject a logger into the interface...

[the remark this answers is not in the document]
Comment (wistefan) [reply src/main/java/org/fiware/iam/TILMapper.java:62]: Fixed it.

File: src/main/java/org/fiware/iam/TILMapper.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+		} catch (IOException ignored) {
+		}
+		try {
+			return OBJECT_READER.readValue(claimValue.getValue(), Boolean.class);
+		} catch (IOException ignored) {
+		}
+		try {
+			return OBJECT_READER.readValue(claimValue.getValue(), Object.class);
+		} catch (IOException e) {
+			return null;
```
[the remark this answers is not in the document]
Comment (wistefan) [reply src/main/java/org/fiware/iam/TILMapper.java:93]: Fixed it.

**Review by pulledtim — APPROVED** [review]

**Comment by beknazaresenbek** [comment]
"mvn install" ends successfully, but at some point in the middle I see these exceptions:
...
`DEBUG i.m.t.j.DataSourceTransactionManager - Initiating transaction rollback on application exception
javax.validation.ConstraintViolationException: existsById.id: must not be null
        at io.micronaut.validation.ValidatingInterceptor.intercept(ValidatingInterceptor.java:111)
        at io.micronaut.aop.chain.MethodInterceptorChain.proceed(MethodInterceptorChain.java:137)
        at org.fiware.iam.repository.H2TrustedIssuerRepository$Intercepted.existsById(Unknown Source)
        at org.fiware.iam.rest.TrustedIssuersListController.createTrustedIssuer(TrustedIssuersListController.java:36)
`
...
`[main] ERROR i.m.r.intercept.RecoveryInterceptor - Type [org.fiware.iam.til.api.IssuerApiTestClient$Intercepted] executed with error: Bad Request
io.micronaut.http.client.exceptions.HttpClientResponseException: Bad Request
`
...
`DEBUG i.m.t.j.DataSourceTransactionManager - Initiating transaction rollback on application exception
org.fiware.iam.exception.ConflictException: Issuer already exists.
        at org.fiware.iam.rest.TrustedIssuersListController.createTrustedIssuer(TrustedIssuersListController.java:37)
`
...
`[main] ERROR i.m.r.intercept.RecoveryInterceptor - Type [org.fiware.iam.til.api.IssuerApiTestClient$Intercepted] executed with error: Client '/': Conflict
io.micronaut.http.client.exceptions.HttpClientResponseException: Client '/': Conflict`
...

If these are just expected messages, then ok

**Comment by wistefan** [comment]
> "mvn install" ends successfully, but at some point in the middle I see these exceptions: ... `DEBUG i.m.t.j.DataSourceTransactionManager - Initiating transaction rollback on application exception javax.validation.ConstraintViolationException: existsById.id: must not be null at io.micronaut.validation.ValidatingInterceptor.intercept(ValidatingInterceptor.java:111) at io.micronaut.aop.chain.MethodInterceptorChain.proceed(MethodInterceptorChain.java:137) at org.fiware.iam.repository.H2TrustedIssuerRepository$Intercepted.existsById(Unknown Source) at org.fiware.iam.rest.TrustedIssuersListController.createTrustedIssuer(TrustedIssuersListController.java:36) ` ... `[main] ERROR i.m.r.intercept.RecoveryInterceptor - Type [org.fiware.iam.til.api.IssuerApiTestClient$Intercepted] executed with error: Bad Request io.micronaut.http.client.exceptions.HttpClientResponseException: Bad Request ` ... `DEBUG i.m.t.j.DataSourceTransactionManager - Initiating transaction rollback on application exception org.fiware.iam.exception.ConflictException: Issuer already exists. at org.fiware.iam.rest.TrustedIssuersListController.createTrustedIssuer(TrustedIssuersListController.java:37) ` ... `[main] ERROR i.m.r.intercept.RecoveryInterceptor - Type [org.fiware.iam.til.api.IssuerApiTestClient$Intercepted] executed with error: Client '/': Conflict io.micronaut.http.client.exceptions.HttpClientResponseException: Client '/': Conflict` ...
> 
> If these are just expected messages, then ok

```mvn install``` does also execute the tests, which include testing for correct error handling in case of conflict and bad requests. When those tests run, the tested code will also log those error, thus the messages are expected.
